client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/site-setup-design-picker.tsx

```diff
@@ -0,0 +1,457 @@
+			setIsForceStaticDesigns( false );
+			return;
+		}
+
+		goBack();
+	};
+
+	const handleShouldStickyChange = ( shouldSticky: boolean ) => {
+		setIsSticky( shouldSticky );
+	};
+
+	// Track scroll event to make sure people are scrolling on mobile.
+	useTrackScrollPageFromTop( isMobile && ! selectedDesign, flow || '', STEP_NAME, {
+		is_generated_designs: showGeneratedDesigns,
+	} );
+
+	// Make sure people is at the top when:
+	// 1. Switching between generated designs and static designs.
+	// 2. Entering/leaving preview mode.
+	useEffect( () => {
+		window.scrollTo( { top: 0 } );
+	}, [ isForceStaticDesigns, !! selectedDesign ] );
+
+	// When the intent is build, we can potentially show the generated design picker.
+	// Don't render until we've fetched the generated designs from the backend.
+	if ( ! site || isLoadingGeneratedDesigns ) {
+		return null;
+	}
+
+	if ( selectedDesign && ! showGeneratedDesigns ) {
+		const isBlankCanvas = isBlankCanvasDesign( selectedDesign );
+		const designTitle = isBlankCanvas ? translate( 'Blank Canvas' ) : selectedDesign.title;
+		const shouldUpgrade = selectedDesign.is_premium && ! isPremiumThemeAvailable;
+		const previewUrl = getDesignPreviewUrl( selectedDesign, {
+			language: locale,
+			// If the user fills out the site title with write intent, we show it on the design preview
+			siteTitle: intent === 'write' ? siteTitle : undefined,
+		} );
+
+		const stepContent = (
+			<WebPreview
+				showPreview
+				showClose={ false }
+				showEdit={ false }
+				externalUrl={ siteSlug }
+				showExternal={ true }
+				previewUrl={ previewUrl }
+				loadingMessage={ translate( '{{strong}}One moment, please…{{/strong}} loading your site.', {
+					components: { strong: <strong /> },
+				} ) }
+				toolbarComponent={ PreviewToolbar }
+				siteId={ site?.ID }
+				url={ site?.URL }
+				translate={ translate }
+				recordTracksEvent={ recordTracksEvent }
+			/>
+		);
+
+		return (
+			<StepContainer
+				stepName={ STEP_NAME }
+				stepContent={ stepContent }
+				hideSkip
+				hideNext={ shouldUpgrade }
+				className={ 'design-setup__preview' }
+				nextLabelText={ translate( 'Start with %(designTitle)s', { args: { designTitle } } ) }
+				goBack={ handleBackClick }
+				goNext={ () => pickDesign() }
+				formattedHeader={
+					<FormattedHeader
+						id={ 'design-setup-header' }
+						headerText={ designTitle }
+						align={ isMobile ? 'left' : 'center' }
+					/>
+				}
+				customizedActionButtons={
+					<>
+						{ shouldUpgrade ? (
+							<Button primary borderless={ false } onClick={ upgradePlan }>
+								{ translate( 'Upgrade Plan' ) }
+							</Button>
+						) : undefined }
+					</>
+				}
+				recordTracksEvent={ recordStepContainerTracksEvent }
+			/>
+		);
+	}
+
+	const heading = (
+		<FormattedHeader
+			id={ 'step-header' }
+			headerText={ headerText() }
+			subHeaderText={ subHeaderText() }
+			align="left"
+		/>
+	);
+
+	const stepContent = showGeneratedDesigns ? (
```

Comment:
What do you think about these methods to remove the large ternary condition from the `stepContent` constant?

**Method 1**
```
let stepContent = (
  <DesignPicker
  ...
  />
)
if ( showGeneratedDesigns ) stepContent = (
  <GeneratedDesignPicker
  ...
  />
)
<DesignPicker
  stepContent={ stepContent }
/>
```

**Method 2**
```
const designPicker = (
  <DesignPicker
  ...
  />
)
const generatedDesignPicker = (
  <GeneratedDesignPicker
  ...
  />
)
<DesignPicker
  stepContent={ showGeneratedDesigns ? generatedDesignPicker : designPicker }
/>
```
These methods or even others could make the code cleaner thanks to removing the large ternary

Reply:
I still prefer the current way but it's just my personal preference. I think maybe creating a function to render the content might be better. In further, I think we could also separate them into different components and share re-usable functions.